test/Marketplace.test.js

```diff
@@ -778,17 +842,48 @@ describe("Marketplace", function () {
       )
       const endBalance = await token.balanceOf(clientWithdrawRecipient.address)
       expect(endBalance - ACCOUNT_STARTING_BALANCE).to.equal(
-        price(request) - expectedPartialhostRewardRecipient
+        maxPrice(request) - expectedPartialhostRewardRecipient
       )
     })
+
+    it.skip("when slot is freed and not filled, client will get refunded the freed slot's funds", async function () {
```

Comment:
🔴 It's not entirely clear to me what requirement you're trying to test here. I'd expect the requirement to be that the client is given the funds that remain after the hosts are paid. So if you know the payout to the hosts, then the rest is for the client. I'd then also structure the test in that way:
1. fill a slot halfway through the expiry window
2. when contract is finished, payout the hosts
3. withdraw funds, and check that it equals the paid amount minus the payouts to the hosts

Reply:
It needs rework agree, but I am trying to be here a bit more explicit in my assertions here, because doing assertions on "something is something minus something" is IMO quite week, because if there is bug in the calculation of the reward, then this would pass and it would not catch the sizing of the propotions...

Also, I am trying to test case where request does finish with not all slots filled and that the reward for that slot is returned.

Reply:
Alright, I'll look at it again after you rework the test.

Reply:
Done, please check again.